test/Kestrel.FunctionalTests/BadHttpRequestTests.cs

```diff
@@ -169,6 +172,86 @@ public Task BadRequestIfHostHeaderDoesNotMatchRequestTarget(string requestTarget
             Assert.Equal(LogLevel.Information, maxLogLevel);
         }
 
+        [Fact]
+        public async Task OverlongHeaderIsRejected()
+        {
+            using (var server = CreateServer())
+            {
+                using (var client = server.CreateConnection())
+                {
+                    await client.Send(
+                        "GET / HTTP/1.1",
+                        "Host:",
+                        "Connection: close");
+
+                    var payload = new string('x', 102400);
+                    client.Socket.Send(Encoding.ASCII.GetBytes(String.Format("Range: {0}\r\n\r\n", payload)));
+
+                    await client.ReceiveStartsWith("HTTP/1.1 431");
+                }
+            }
+        }
+
+        [Fact]
+        public async Task HostHeaderAcceptedWithPeriod()
+        {
+            using (var server = CreateServer())
+            {
+                using (var client = server.CreateConnection())
+                {
+                    await client.Send(
+                        "GET / HTTP/1.1",
+                        "Host: .",
+                        "Connection: close\r\n\r\n");
+
+                    await client.ReceiveStartsWith("HTTP/1.1 200");
+                }
+            }
+        }
+
+        [Fact]
+        public async Task UnprintableCharactersAreFiltered()
+        {
+            using (var server = CreateServer())
+            {
+                using (var client = server.CreateConnection())
+                {
+                    await client.Send(
+                        "GET /\x0D\0x0ALocation:http://www.contoso.com/ HTTP/1.1",
+                        "Host:\r\n\r\n");
```

Comment:
Agreed. I can adjust the name to more accurately reflect the bug title. Note that this test was copied directly from the repro steps in the bug (TFS-221148). I didn't file the bug originally so I didn't see the issue for myself. (I did try and after many attempts could not get old version of Kestrel to build.)

Now that I'm taking another look, the use of Location as a request header seems non-sensical but I'm not sure the header in and of itself is relevant. My understanding of Kestrel's behavior when this bug was discovered was that Kestrel would a) accept the split request line, b) accept the injected header, and c) return HTTP 200.

In any event, the issue doesn't repro now given the steps provided in the bug, which is what the test is asserting.